### ISSUE TITLE
[Feature Request] API - Add hasFile from MovieResource to CollectionMovieResource

### ISSUE THREAD

**Issue by dynamiccookies** [Radarr/Radarr#7869]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Not exactly - It would just be nice to have

### Describe the solution you'd like

Please add the hasFile attribute from MovieResource to the CollectionMovieResource in the API

![MovieResource](https://user-images.githubusercontent.com/9450183/206764618-e65c6653-da1a-42cb-8e8e-eace20a9fd36.png)
![CollectionMovieResource](https://user-images.githubusercontent.com/9450183/206764647-6baa3e9e-3951-4f47-a497-c3d1393ee438.png)

### Describe alternatives you've considered

The only alternative I can think of is to check each movie returned in a collection against /api/v3/movie using the tmdbId. Unfortunately, that will require many extra unnecessary calls to the API just to see if the movie returned in the collection call currently exists in the library.

### Anything else?

I can't think of anything else to add, but please feel free to let me know if you have any questions. Thank you!

[AB#4395](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/4395)

**Comment by austinwbest** [comment]
I think for future reasons, `sizeOnDisk` might be better than `hasFile` but Q can confirm that.

**Comment by dynamiccookies** [comment]
> I think for future reasons, `sizeOnDisk` might be better than `hasFile` but Q can confirm that.

Absolutely, that would work too. Then I could just check to see if it's greater than 0.

**Comment by Qstick** [comment]
Correct, `hasFile` is disappearing in v5 so it won't make it.

**Comment by dynamiccookies** [comment]
> Correct, `hasFile` is disappearing in v5 so it won't make it.

Thank you, that's very good to know. I'm using it in another area of the code I'm working on, so I can be proactive and switch it for `sizeOnDisk`. It makes perfect sense to use that for determining whether or not the file exists on the server. I hope it can be added to the CollectionMovieResource as well.
